### ISSUE TITLE
[BUG?] Legend goes crazy if plotting more than 24 Vertical Lines?

### ISSUE THREAD

**Issue by andrethrill** [ioam/holoviews#2339]
I'm pasting here full code for reproducibility. Please jump to bottom section `Problem Description and Questions` first, if you wish.

# Imports

```python
import numpy as np
import pandas as pd
import datetime
import random
import holoviews as hv
hv.extension('bokeh')
```

# Generation and plotting functions

```python
#generate dummy df
def generate_df(size):
    d = {
        'dates':
        pd.date_range('1980-01-01', periods=size, freq='1T'),
        'signal1':
        np.random.normal(0, 0.3, size=size).cumsum() + 50,
        'signal2':
        np.random.normal(0, 0.3, size=size).cumsum() + 50,
        'signal3':
        np.random.normal(0, 0.3, size=size).cumsum() + 50
    }

    df = pd.DataFrame(d)

    return df.melt(id_vars='dates').sort_values('dates')
```

```python
#generate dummy dates
def generate_random_date(begin, end):
    
    diff = end - begin
    
    
    return begin + datetime.timedelta(seconds=random.randint(0, int(diff.total_seconds())))

def generate_random_dates(begin, end, n):
    
    return [generate_random_date(begin, end) for i in range(0,n)]
```

```python
#plot a timeseries
def plot_timeseries(df, width=800, height=400, title='plot', legend_position='top_right'):

        plot_opts=dict()
        plot_opts['show_grid'] = True
        plot_opts['width'] = width
        plot_opts['height'] = height
        plot_opts['legend_position'] = legend_position
        plot_opts['legend_limit'] = 1000


        curve_1 = hv.Curve( df,
                     kdims=['dates'],
                     vdims=['value', 'variable'],
                     label=title).groupby('variable')


        return curve_1.overlay().opts(plot=plot_opts)
```

```python
#Overlay a list of vertical lines
def vertical_lines(vlines_list):

        plot_opts=dict()
        plot_opts['show_legend'] = False
        plot_opts['show_title'] = False
        
        style_opts = dict()
        style_opts['line_color'] = 'brown'
        style_opts['linewidth'] = '4'
        
        
        vlines = []
        for line in vlines_list:
            vlines.append(hv.VLine(line)(plot=plot_opts,
                                         style=style_opts))

        return hv.Overlay(vlines)
```

# Generate df and dates and plot them

```python
small_df = generate_df(10000)
small_df.head()
```

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>dates</th>
      <th>variable</th>
      <th>value</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1980-01-01 00:00:00</td>
      <td>signal1</td>
      <td>49.800754</td>
    </tr>
    <tr>
      <th>20000</th>
      <td>1980-01-01 00:00:00</td>
      <td>signal3</td>
      <td>50.292620</td>
    </tr>
    <tr>
      <th>10000</th>
      <td>1980-01-01 00:00:00</td>
      <td>signal2</td>
      <td>50.425544</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1980-01-01 00:01:00</td>
      <td>signal1</td>
      <td>49.896840</td>
    </tr>
    <tr>
      <th>20001</th>
      <td>1980-01-01 00:01:00</td>
      <td>signal3</td>
      <td>50.497810</td>
    </tr>
  </tbody>
</table>
</div>

## With 24 lines

```python
vlines_list = generate_random_dates(small_df.dates.min(), small_df.dates.max(), 24)


vertical_lines(vlines_list) * plot_timeseries(small_df, title='24 lines')
```

![unknown](https://user-images.githubusercontent.com/25300892/36160205-822a19f8-10d8-11e8-80d7-601f5755fb98.png)

## With 25 lines

```python
vlines_list = generate_random_dates(small_df.dates.min(), small_df.dates.max(), 25)


vertical_lines(vlines_list) * plot_timeseries(small_df, title='25 lines')
```

![unknown](https://user-images.githubusercontent.com/25300892/36160218-8dc977a4-10d8-11e8-9b43-438581b11500.png)

# Problem Description and Questions

I am overlaying a Curves Overlay (function `plot_timeseries`) with many vertical lines (function `vertical_lines`).

If I generate and plot 24 vertical lines the legend appears **correctly** (see legend of figure section `With 24 lines`).
If I generate and plot 25 vertical lines the legend appears **incorrectly** (see legend of figure section `With 25 lines`).

Questions:

1. Is this a bug?
2. Or am I doing something wrong? For instance, is there a more efficient way of plotting several vertical lines besides calling VLine repeatedly? Such as, sending a list as argument.

**Comment by philippjfr** [comment]
Legends are limited to 25 entries by default because things often getting messy otherwise. Since VLine does not actually have a legend entry it shouldn't count toward this limit but seemingly it does, so I'll leave this open. For now you raise the limit with:

``overlay.opts(plot=dict(legend_limit=50))``

or do so globally:

```python
hv.opts({'Overlay': {'plot': dict(legend_limit=50)}})
```

**Comment by andrethrill** [comment]
Thanks @philippjfr  that solved the problem!

But I do agree that the VLines in my example above should not enter to the legend entries count limit.

**Comment by neighthan** [comment]
Is there a reason why `VLine` does not have a legend entry? In this particular example, they don't want the vlines appearing in the legend, but even if I pass in `label="vline"`, I can't get the vline to appear in the legend. Should I `Spikes` instead of `VLine` if I need a legend entry?

**Comment by philippjfr** [comment]
Yes, VLine/HLine aren't likely to gain support legends any time soon. The reason is that in bokeh they are implemented as a ``Span`` which is an Annotation, and unlike glyphs those don't take part in the generated legend.

**Comment by neighthan** [comment]
Is that mentioned in the docs somewhere? Not necessarily the specific reason, but at least that `VLine`/`HLine` can't appear in legends? I'll just use `Spikes` instead, and that will work fine for me, but I thought there was some bug with the legend before.

**Comment by philippjfr** [comment]
Probably not, deserves a little note in the VLine/HLine reference notebooks for bokeh (and maybe matplotlib too) as I'm not sure axvline/axhline participate in legends there either.
